botocore/utils.py

```diff
@@ -1569,6 +1571,45 @@ def set_endpoint(self, request, **kwargs):
     def _use_accesspoint_endpoint(self, request):
         return 's3_accesspoint' in request.context
 
+    def _validate_fips_supported(self, request):
+        client_region = request.context.get('client_region', '')
+        if 'fips' not in client_region:
+            return
+        if 'outpost_name' in request.context['s3_accesspoint']:
+            raise UnsupportedS3AccesspointConfigurationError(
+                msg=(
+                    'Invalid configuration Outpost Access Points '
+                    'do not support Fips- regions'
+                )
+            )
+        client_region = client_region.replace('fips-', '')
```

Comment:
It may be good to look for the `<region>-fips` pattern as well. It seems like some fips regions can be defined in that fashion and looks like the Ruby implementation is doing that as well: https://github.com/aws/aws-sdk-ruby/pull/2495/files#diff-10a2f251424fdb0be98d76e54c194c4d2eeebedb018faab5db57c22ffbc92a61R70

Reply:
Yes, but since `endpoints.json` doesn't have such entry for S3 it won't work  in any case